tests/tools/test_shell.py

```diff
@@ -47,6 +49,41 @@
     last line
 """
 
+if HAS_PEXPECT:
+
+    class AnsiExpecter(pexpect.expect.Expecter):
```

Comment:
Maybe stupid question, but can we use `unittest.mock.patch` (https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) to temporarily patch `pexpect` with this behavior?